### PR TITLE
Document pre-release testing process and update release docs

### PR DESCRIPTION
**Pull request by markdboyd**

### CODE REVIEW

File: bamboo/create-release.sh [hunk 1]
```diff
@@ -5,26 +5,19 @@ source .bamboo_env_vars || true
 . ./bamboo/set-bamboo-env-variables.sh
 . ./bamboo/abort-if-not-pr.sh
 
+# Build TF module artifacts
+. ./bamboo/create-release-artifacts.sh
```
Comment (markdboyd) [inline bamboo/create-release.sh:9]: @Jkovarik This should work, right? Not sure how to test this before actually doing a release

File: docs/development/release.md [hunk 2]
```diff
@@ -6,6 +6,45 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+      # for data-persistence
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/data-persistence"
+
+      # for cumulus
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/cumulus"
+      ```
+
+   2. Run `terraform init`
+   3. Run `terraform apply`
+8. Follow any migration steps noted in `CHANGELOG.md` and confirm that they are successful
```
Comment (markdboyd) [inline docs/development/release.md:46]: Is this sufficient? I wasn't sure what else to mention here, if anything.

I suppose that there are migration steps that you might need to do **before** deploying the latest version, but we don't clearly call that out in our CHANGELOG often

File: bamboo/create-release-artifacts.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+set -ex
+
+## Build TF modules that require source building
+
+(cd tf-modules/distribution && ./bin/build-tf-module.sh && cp ./dist/terraform-aws-cumulus-distribution.zip ../../terraform-aws-cumulus-distribution.zip)
+(cd tf-modules/s3-replicator && ./bin/build-tf-module.sh && cp ./dist/terraform-aws-cumulus-s3-replicator.zip ../../terraform-aws-cumulus-s3-replicator.zip)
+
+## Create zipfiles
+zip -FS -r -x \*node_modules\* \*internal\* -o terraform-aws-cumulus.zip tf-modules lambdas tasks/**/dist/** packages/**/dist/**
```
Comment (botanical) [inline bamboo/create-release-artifacts.sh:11]: What is the difference between file syncing and update?

File: bamboo/create-release.sh [hunk 4]
```diff
@@ -5,26 +5,19 @@ source .bamboo_env_vars || true
...
 
-## Build TF modules that require source building
-
-(cd tf-modules/distribution && ./bin/build-tf-module.sh && cp ./dist/terraform-aws-cumulus-distribution.zip ../../terraform-aws-cumulus-distribution.zip)
-(cd tf-modules/s3-replicator && ./bin/build-tf-module.sh && cp ./dist/terraform-aws-cumulus-s3-replicator.zip ../../terraform-aws-cumulus-s3-replicator.zip)
-
-## Create zipfiles
-zip -FS -r -x \*node_modules\* \*internal\* -o terraform-aws-cumulus.zip tf-modules lambdas tasks/**/dist/** packages/**/dist/**
-(cd ./tf-modules/workflow; zip -FS -r -x \*node_modules\* -o ../../terraform-aws-cumulus-workflow.zip .)
-(cd ./tf-modules/cumulus_ecs_service; zip -FS -r -x \*node_modules\* -o ../../terraform-aws-cumulus-ecs-service.zip .)
```
Comment (botanical) [inline bamboo/create-release.sh:26]: Why did you remove these?

File: docs/development/release.md [hunk 5]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
 
 Read more about the semantic versioning [here](https://docs.npmjs.com/getting-started/semantic-versioning).
 
+## Pre-release testing
+
+Before releasing a new major version of Cumulus that contains breaking changes, we should test the deployment upgrade from the latest stable release of Cumulus to the upcoming release with breaking changes.
+
+It is preferable to use the [`cumulus-template-deploy`](`https://github.com/nasa/cumulus-template-deploy`) repo for testing the deployment, since this is the officially recommended deployment configuration for the users.
+
+1. Checkout the `cumulus-template-deploy` repo
```
Comment (botanical) [inline docs/development/release.md:15]:
```suggestion
1. Checkout the [`cumulus-template-deploy`](`https://github.com/nasa/cumulus-template-deploy`) repo
```

File: docs/development/release.md [hunk 6]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+2. Update the deployment code to use the latest stable release artifacts if it hasn't already been updated:
```
Comment (botanical) [inline docs/development/release.md:16]: How does one check if it hasn't already been updated?

File: bamboo/create-release.sh (hunk 1 above)
Comment (Jkovarik) [reply bamboo/create-release.sh:9]: We can test the bash is good by running it in the container bamboo would be running it in, if we're worried about cross mount perms (this *probably isn't an issue) we could mock something up in bamboo, but that's probably not needed.

Comment (Jkovarik) [reply bamboo/create-release.sh:9]: You need to make sure the permissions on the referenced script have u+x in the repo though.

File: docs/development/release.md [hunk 7]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+    ```text
+    # for data-persistence
+    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip//tf-modules/data-persistence"
```
Comment (botanical) [inline docs/development/release.md:20]:
```suggestion
    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip/tf-modules/data-persistence"
```

File: docs/development/release.md [hunk 8]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+    # for cumulus
+    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip//tf-modules/cumulus"
```
Comment (botanical) [inline docs/development/release.md:23]:
```suggestion
    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip/tf-modules/cumulus"
```

File: docs/development/release.md [hunk 9]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+3. For both the `data-persistence-tf` and `cumulus-tf` modules:
+   1. Add the necessary backend configuration (`terraform.tf`) and variables (`terraform.tfvars`)
+   2. Run `terraform init`
+   3. Run `terraform apply`
+4. Checkout the `master` branch of the `cumulus` repo
+5. Run a full bootstrap of the code: `npm run bootstrap`
+6. Build the release artifacts: `./bamboo/create-release-artifacts.sh`
+7. For both the `data-persistence-tf` and `cumulus-tf` modules:
+   1. Update the deployment to use the built release artifacts:
```
Comment (botanical) [inline docs/development/release.md:34]: It's not obvious to me where you update the deployment to use the built release artifacts. Not sure if I am alone in this, but do you mind elaborating?

File: docs/development/release.md [hunk 10]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+      ```text
+      # for data-persistence
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/data-persistence"
```
Comment (botanical) [inline docs/development/release.md:38]:
```suggestion
      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip/tf-modules/data-persistence"
```

File: docs/development/release.md [hunk 11]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+      # for cumulus
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/cumulus"
```
Comment (botanical) [inline docs/development/release.md:41]:
```suggestion
      source = "/Users/[USER]/development/cumulus/terraform-aws-cumulus.zip/tf-modules/cumulus"
```

File: docs/development/release.md [hunk 12]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
 
 Read more about the semantic versioning [here](https://docs.npmjs.com/getting-started/semantic-versioning).
 
+## Pre-release testing
```
Comment (Jkovarik) [inline docs/development/release.md:9]: I feel like we should consider finding a way to automate this, although that's probably dredging up the 'should we integration test the template discussion'.

File: docs/development/release.md [hunk 13]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+
+      ```text
+      # for data-persistence
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/data-persistence"
+
+      # for cumulus
+      source = "/Users/mboyd/development/cumulus/terraform-aws-cumulus.zip//tf-modules/cumulus"
+      ```
+
+   2. Review the `CHANGELOG.md` for any pre-deployment migration steps and confirm that they are successful
```
Comment (botanical) [inline docs/development/release.md:44]: How does one confirm they are successful?

File: docs/development/release.md (hunk 10 above)
Comment (Jkovarik) [inline docs/development/release.md:38]: Consider removing the user-specific path here

File: docs/development/release.md [hunk 14]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
 #### From Master
 
-If creating a new minor version release from master, create a branch titled `release-MAJOR.MINOR.x` (e.g. `release-1.14.x`) as a minor version base branch from master to allow us to easily backport patches to that version. Push the `release-MAJOR.MINOR.x` branch to GitHub if it was created locally. (Commits should be even with master at this point.)
+Create a branch titled `release-MAJOR.MINOR.x` for the release.
+
+If creating a new major version release from master, say `5.0.0`, then the branch would be named `release-5.0.x`. If creating a new minor version release from master, say `1.14.0` then the branch would be named `release-1.14.x`.
+
+Having a release branch for each major/minor version allows us to easily backport patches to that version.
+
+Push the `release-MAJOR.MINOR.x` branch to GitHub if it was created locally. (Commits should be even with master at this point.)
```
Comment (botanical) [inline docs/development/release.md:61]: Love this section 💯

File: docs/development/release.md [hunk 15]
```diff
@@ -24,8 +70,14 @@ git checkout -b release-1.14.0
...
-When creating a backport, a minor version base branch should already exist on GitHub.
-Check out the existing minor version base branch then create a release branch from it.
+When creating a backport, a minor version base branch should already exist on GitHub. Check out the existing minor version base branch then create a release branch from it. For example:
+
+```bash
+# check out existing minor version base branch
+git checkout release-1.14.x
+# create new release branch for backport
+git checkout -b release-1.14.1
+```
```
Comment (botanical) [inline docs/development/release.md:80]: This is awesome

File: docs/development/release.md [hunk 16]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
 
 Read more about the semantic versioning [here](https://docs.npmjs.com/getting-started/semantic-versioning).
 
+## Pre-release testing
+
+Before releasing a new major version of Cumulus that contains breaking changes, we should test the deployment upgrade from the latest stable release of Cumulus to the upcoming release with breaking changes.
```
Comment (Jkovarik) [inline docs/development/release.md:11]: Wasn't our intention here to be PI-stable -> PI stable, not all major versions?

File: docs/development/release.md [hunk 17]
```diff
@@ -24,8 +70,14 @@ git checkout -b release-1.14.0
 
 #### Backporting
 
-When creating a backport, a minor version base branch should already exist on GitHub.
-Check out the existing minor version base branch then create a release branch from it.
+When creating a backport, a minor version base branch should already exist on GitHub. Check out the existing minor version base branch then create a release branch from it. For example:
```
Comment (Jkovarik) [inline docs/development/release.md:73]: 👍🏻

File: docs/development/release.md [hunk 18]
```diff
@@ -72,59 +124,66 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
 git add CONTRIBUTORS.md
 ```
 
+Commit and push these changes, if any.
```
Comment (botanical) [inline docs/development/release.md:127]:
```suggestion
Commit and push these changes, if any.

```bash
git commit -m "Update CONTRIBUTORS.md"
git push release-MAJOR.MINOR.x
```

```

This is a nit

File: docs/development/release.md [hunk 19]
```diff
@@ -72,59 +124,66 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
...
+git tag -a v1.x.x -m "Release 1.x.x"
+git push origin v1.x.x
 ```
 
 ### 10. Running the deployment
 
 Publishing of new releases is handled by a Bamboo release plan and is manually triggered.
 
-If you created a new release plan in step one, you will need to create a new bamboo deployment plan
+If you created a new release plan in step one, you will need to create a new bamboo deployment plan following the instructions below.
```
Comment (Jkovarik) [inline docs/development/release.md:186]:
```suggestion
If you created a new release plan in step one, you will need to create a new bamboo deployment plan following the instructions below:
```

File: docs/development/release.md [hunk 20]
```diff
@@ -157,7 +216,7 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
 
 ### 11. Create a new Cumulus release on github
 
-The CI release scripts will automatically create a release based on the release version tag, as well as uploading release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
+The CI release scripts will automatically create a Github release based on the release version tag, as well as uploading release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```
Comment (Jkovarik) [inline docs/development/release.md:219]:
```suggestion
The CI release scripts will automatically create a Github release based on the release version tag, as well as uploading artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```

**Review by Jkovarik — CHANGES_REQUESTED** [review]
👍🏻  just a couple of minor nits/comments to address/comment on.

File: docs/development/release.md [hunk 21]
```diff
@@ -72,59 +124,66 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
...
-Note: This is for 1.10.3 or later.
+Commit and push these changes.
 
 ### 8. Create a pull request against the minor version branch
 
 1. Push the release branch (e.g. `release-1.2.3`) to GitHub.
-2. Create a PR against the minor version base branch  e.g. `release-1.2.x`).
-3. Configure CI to run automated tests against this PR by finding the branch plan for the release branch and setting these variables:
+2. Create a PR against the minor version base branch (e.g. `release-1.2.x`).
+3. Configure CI to run automated tests against this PR by finding the branch plan for the release branch (`release-1.2.3`) and setting these variables:
```
Comment (botanical) [inline docs/development/release.md:159]: Do we set these variables in Bamboo?

File: docs/development/release.md (hunk 20 above)
Comment (botanical) [inline docs/development/release.md:219]:
```suggestion
The CI release scripts will automatically create a Github release based on the release version tag, as well as upload release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```

File: bamboo/create-release.sh (hunk 4 above)
Comment (markdboyd) [reply bamboo/create-release.sh:26]: I moved them to the new `create-release-artifacts.sh` script, because when you want to do local testing of the next release, you want to run that script to create the artifacts locally without the rest of the behavior of the `create-release.sh` script

File: docs/development/release.md (hunk 7 above)
Comment (markdboyd) [reply docs/development/release.md:20]: It's weird, but in the case of the release artifacts, the `//` is actually correct. The reason why is that the full `terraform-aws-cumulus.zip` contains all of our Terraform modules so that relative paths between them are supported (for example https://github.com/nasa/cumulus/blob/master/tf-modules/cumulus/archive.tf#L2). So we have one zipfile for both data-persistence and cumulus and you have to reference them directly in the zipfile using this syntax.

See https://www.terraform.io/docs/language/modules/sources.html#modules-in-package-sub-directories

File: docs/development/release.md (hunk 8 above)
Comment (markdboyd) [reply docs/development/release.md:23]: See above comment

File: docs/development/release.md (hunk 13 above)
Comment (markdboyd) [reply docs/development/release.md:44]: It's a bit hard to be more specific given the variance in what "migration steps" can be. Is there any way you think this could be worded more clearly though?

File: docs/development/release.md (hunk 16 above)
Comment (markdboyd) [reply docs/development/release.md:11]: I guess by "stable" I didn't mean PI stable, though I can see the confusion. I was just saying that if you're releasing a new major major version (6.0.0), we need to at least be sure that the upgrade path from the latest release (5.0.1) is successful. For each PI, we may also need to guarantee that going from last PI release to new PI release is successful, but that's not what I was trying to discuss here.

Would this help (just removing the word "stable" to avoid confusion)?

```suggestion
Before releasing a new major version of Cumulus that contains breaking changes, we should test the deployment upgrade from the latest release of Cumulus to the upcoming release with breaking changes.
```

File: docs/development/release.md (hunk 21 above)
Comment (markdboyd) [reply docs/development/release.md:159]: Yeah, I can update that to be more clear

File: bamboo/create-release-artifacts.sh (hunk 3 above)
Comment (markdboyd) [reply bamboo/create-release-artifacts.sh:11]: I'm not sure I understand what you're specifically asking about

File: docs/development/release.md (hunk 6 above)
Comment (markdboyd) [reply docs/development/release.md:16]: You have to look in the deployment code to see whether the source for data-persistence and cumulus modules already reference the latest release. Suggestions welcome on how to make this more clear, but I can try to update it

File: docs/development/release.md (hunk 10 above)
Comment (markdboyd) [reply docs/development/release.md:38]: Going to resolve this in favor of Jon's suggestion to not assume people have the same local path for cumulus that I do

File: docs/development/release.md (hunk 16 above)
Comment (Jkovarik) [reply docs/development/release.md:11]: I think that updated wording makes the intent more clear - I'm not (yet) convinced that in all cases this test makes sense, particularly given a per-sprint cadence but that's a discussion we can have offline (arch meeting).

File: docs/development/release.md (hunk 10 above)
Comment (Jkovarik) [reply docs/development/release.md]: This was updated here:  c03c79d753e08e325907beca0d99724abb87d3c4

Comment (Jkovarik) [reply docs/development/release.md]: c03c79d753e08e325907beca0d99724abb87d3c4

File: docs/development/release.md (hunk 19 above)
Comment (Jkovarik) [reply docs/development/release.md]: https://github.com/nasa/cumulus/pull/2087/commits/c03c79d753e08e325907beca0d99724abb87d3c4

File: docs/development/release.md (hunk 20 above)
Comment (Jkovarik) [reply docs/development/release.md]: https://github.com/nasa/cumulus/pull/2087/commits/c03c79d753e08e325907beca0d99724abb87d3c4

File: docs/development/release.md (hunk 7 above)
Comment (botanical) [reply docs/development/release.md:20]: Oh interesting!

File: docs/development/release.md (hunk 9 above)
Comment (botanical) [reply docs/development/release.md:34]: Okay, I think I understand now so you can ignore this

File: docs/development/release.md [hunk 22]
```diff
@@ -6,13 +6,59 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+8. Review the `CHANGELOG.md` for any post-deployment migration steps and confirm that they are successful
+
 ## Updating Cumulus version and publishing to NPM
 
 ### 1. Create a branch for the new release
 
 #### From Master
 
-If creating a new minor version release from master, create a branch titled `release-MAJOR.MINOR.x` (e.g. `release-1.14.x`) as a minor version base branch from master to allow us to easily backport patches to that version. Push the `release-MAJOR.MINOR.x` branch to GitHub if it was created locally. (Commits should be even with master at this point.)
+Create a branch titled `release-MAJOR.MINOR.x` for the release.
```
Comment (botanical) [inline docs/development/release.md:55]:
```bash
git checkout -b release-MAJOR.MINOR.x
```

For consistency with what we write below

File: docs/development/release.md (hunk 13 above)
Comment (botanical) [reply docs/development/release.md:44]: Hm, maybe:
```suggestion
   2. Review the `CHANGELOG.md` for any pre-deployment migration steps. If there are, go through the steps and confirm that they are successful
```

File: docs/development/release.md (hunk 6 above)
Comment (markdboyd) [reply docs/development/release.md]: Is this better?

35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md (hunk 13 above)
Comment (markdboyd) [reply docs/development/release.md]: 35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md (hunk 16 above)
Comment (markdboyd) [reply docs/development/release.md]: 35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md (hunk 21 above)
Comment (markdboyd) [reply docs/development/release.md]: 35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md (hunk 20 above)
Comment (markdboyd) [reply docs/development/release.md]: 35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md (hunk 22 above)
Comment (markdboyd) [reply docs/development/release.md:59]: 35d6eb91364027eb7e5441a4823817672a3d8223

File: docs/development/release.md [hunk 23]
```diff
@@ -157,7 +235,7 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
 
 ### 11. Create a new Cumulus release on github
 
-The CI release scripts will automatically create a release based on the release version tag, as well as uploading release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
+The CI release scripts will automatically create a Github release based on the release version tag, as well as upload artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```
Comment (botanical) [inline docs/development/release.md:238]:
```suggestion
The CI release scripts will automatically create a GitHub release based on the release version tag, as well as upload artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```

File: docs/development/release.md [hunk 24]
```diff
@@ -72,81 +132,99 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
...
+### 10. Publishing the release
 
-Publishing of new releases is handled by a Bamboo release plan and is manually triggered.
+Publishing of new releases is handled by a custom Bamboo branch plan and is manually triggered.
 
-If you created a new release plan in step one, you will need to create a new bamboo deployment plan
+The reasons for using a separate branch plan to handle releases instead of the branch plan for the minor version (e.g. `release-1.2.x`) are:
 
-#### Creating a Bamboo Deployment plan
+* The Bamboo build for the minor version release branch is trigged **automatically** on any commits to that branch, whereas we want to manually control when the release is published.
```
Comment (botanical) [inline docs/development/release.md:199]:
```suggestion
* The Bamboo build for the minor version release branch is triggered **automatically** on any commits to that branch, whereas we want to manually control when the release is published.
```

File: bamboo/create-release-artifacts.sh (hunk 3 above)
Comment (Jkovarik) [reply bamboo/create-release-artifacts.sh:11]:
```
copy (-U)
Select entries in an existing archive and copy them to a new archive. This new mode is similar to update but command line patterns select entries in the existing archive rather than files from the file system and it uses the --out option to write the resulting archive to a new file rather than update the existing archive, leaving the original archive unchanged.
The new File Sync option (-FS) is also considered a new mode, though it is similar to update. This mode synchronizes the archive with the files on the OS, only replacing files in the archive if the file time or size of the OS file is different, adding new files, and deleting entries from the archive where there is no matching file. As this mode can delete entries from the archive, consider making a backup copy of the archive.

```

File: docs/development/release.md [hunk 25]
```diff
@@ -6,13 +6,67 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+    ```text
+    # in data-persistence-tf/main.tf
+    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip//tf-modules/data-persistence"
+
+    # in cumulus-tf/main.tf
+    source = "https://github.com/nasa/cumulus/releases/download/v5.0.1/terraform-aws-cumulus.zip//tf-modules/cumulus"
+    ```
+
+3. For both the `data-persistence-tf` and `cumulus-tf` modules:
+   1. Add the necessary backend configuration (`terraform.tf`) and variables (`terraform.tfvars`)
```
Comment (Jkovarik) [inline docs/development/release.md:31]: Should we add commentary here re: *not* using a dev stack?

File: docs/development/release.md [hunk 26]
```diff
@@ -6,13 +6,67 @@ We use a global versioning approach, meaning version numbers in cumulus are cons
...
+   2. Run `terraform init`
+   3. Run `terraform apply`
+4. Checkout the `master` branch of the `cumulus` repo
+5. Run a full bootstrap of the code: `npm run bootstrap`
+6. Build the pre-release artifacts: `./bamboo/create-release-artifacts.sh`
```
Comment (Jkovarik) [inline docs/development/release.md:36]: :100:

File: docs/development/release.md [hunk 27]
```diff
@@ -72,81 +132,99 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
...
-  npm run docs-build-packages
+npm run docs-build-packages
 ```
 
+Commit and push these changes, if any.
+
 ### 7. Cut new version of Cumulus Documentation
 
-If this is a backport, do not version documentation. For various reasons, we do not merge backports back to master, other than changelog notes. Doc changes will not be published to our documentation website.
+> Note: This is for version `1.10.3` or later.
```
Comment (Jkovarik) [inline docs/development/release.md:149]: Is the 1.10.3 comment *useful* at this point?        I can't imagine needing to cut a release < 1.10.3.

File: docs/development/release.md [hunk 28]
```diff
@@ -72,81 +132,99 @@ Similar to #4, make sure the DATA_MODEL_CHANGELOG is updated if there are data m
...
 1. Push the release branch (e.g. `release-1.2.3`) to GitHub.
-2. Create a PR against the minor version base branch  e.g. `release-1.2.x`).
-3. Configure CI to run automated tests against this PR by finding the branch plan for the release branch and setting these variables:
+2. Create a PR against the minor version base branch (e.g. `release-1.2.x`).
+3. Configure Bamboo to run automated tests against this PR by finding the branch plan for the release branch (`release-1.2.3`) and setting only these variables:
 
     * `GIT_PR`: `true`
     * `SKIP_AUDIT`: `true`
 
+    **IMPORTANT**: Do NOT set the `PUBLISH_FLAG` variable to `true` for this branch plan. The actual publishing of the release will be handled by a separate, manually triggered branch plan.
```
Comment (Jkovarik) [inline docs/development/release.md:172]: 👍🏻

**Review by Jkovarik — APPROVED** [review]
The current changeset looks good, a couple of comments for discussion, but I'm comfortable with a merge.
